pkg/operator/operator_test.go

```diff
@@ -228,11 +240,59 @@ func TestOperatorSyncClusterAPIControllerHealthCheckController(t *testing.T) {
 				Platform: v1.AWSPlatformType,
 			},
 		}
-
 		stopCh := make(<-chan struct{})
-		optr := newFakeOperator(nil, []runtime.Object{tc.featureGate, infra}, stopCh)
-		go optr.Run(2, stopCh)
+		optr := newFakeOperator(nil, []runtime.Object{infra}, stopCh)
+		err := initFeatureGateInformer([]runtime.Object{tc.featureGate}, optr)
+		if err != nil {
+			t.Errorf("Failed to set featuregate %v : %v", tc.featureGate, err)
```

Comment:
Isn't this fatal?

Reply:
with `Errorf` this test will be reported failed but others will be executed. Is not that we want?